datetime/datetime_test.go

```diff
@@ -0,0 +1,543 @@
+
+	tuple := Tuple{Cid: 13, Orig: "orig", Events: []Event{{dt1, "Minsk"}, {dt2, "Moscow"}}}
+	resp, err := conn.Replace("testDatetime_3", &tuple)
+	if err != nil || resp.Code != 0 {
+		t.Fatalf("Failed to replace: %s", err.Error())
+		return
+	}
+	if len(resp.Data) != 1 {
+		t.Fatalf("Response Body len != 1")
+	}
+	if tpl, ok := resp.Data[0].([]interface{}); !ok {
+		t.Errorf("Unexpected body of Replace")
+	} else {
+		if len(tpl) != 3 {
+			t.Fatalf("Unexpected body of Replace (tuple len)")
+		}
+		if id, ok := tpl[0].(uint64); !ok || id != 13 {
+			t.Fatalf("Unexpected body of Replace (13)")
+		}
+		if o, ok := tpl[1].(string); !ok || o != "orig" {
+			t.Fatalf("Unexpected body of Replace (orig)")
+		}
+	}
+
+}
+
+func TestCustomEncodeDecodeTuple2(t *testing.T) {
+	if isDatetimeSupported == false {
+		t.Skip("Skipping test for Tarantool without datetime support in msgpack")
+	}
+
+	conn := connectWithValidation(t)
+	defer conn.Close()
+
+	// Setup: insert a value.
+	tm := time.Unix(0, 0)
+	_, err := conn.Insert(space, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime insert failed: %s", err.Error())
+	}
+
+	var tuples []Tuple
+	err = conn.SelectTyped("testDatetime_2", index, 0, 1, IterEq, []interface{}{1, tm}, &tuples)
+	if err != nil {
+		t.Fatalf("Failed to SelectTyped(): %s", err.Error())
+	}
+
+	// Teardown: delete a value.
+	_, err = conn.Delete(space, index, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime delete failed: %s", err.Error())
+	}
+}
+
+func TestCustomEncodeDecodeTuple3(t *testing.T) {
+	if isDatetimeSupported == false {
+		t.Skip("Skipping test for Tarantool without datetime support in msgpack")
+	}
+
+	conn := connectWithValidation(t)
+	defer conn.Close()
+
+	// Setup: insert a value.
+	tm := time.Unix(0, 0)
+	_, err := conn.Insert(space, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime insert failed: %s", err.Error())
+	}
+
+	var tuples2 []Tuple2
+	err = conn.SelectTyped("testDatetime_2", index, 0, 1, IterEq, []interface{}{1, tm}, &tuples2)
+	if err != nil {
+		log.Fatalf("Failed to SelectTyped: %s", err.Error())
+		return
+	}
+
+	// Teardown: delete a value.
+	_, err = conn.Delete(space, index, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime delete failed: %s", err.Error())
+	}
+}
+
+func TestCustomDecodeFunction(t *testing.T) {
+	if isDatetimeSupported == false {
+		t.Skip("Skipping test for Tarantool without datetime support in msgpack")
+	}
+
+	conn := connectWithValidation(t)
+	defer conn.Close()
+
+	// Call function 'call_me_maybe' returning a table of custom tuples.
+	var tuples []Tuple
+	err := conn.CallTyped("call_me_maybe", []interface{}{1}, &tuples)
+	if err != nil {
+		t.Fatalf("Failed to CallTyped: %s", err.Error())
+	}
+
+	if cid := tuples[0].Cid; cid != 5 {
+		t.Fatalf("Wrong Cid (%d), should be 5", cid)
+	}
+	if orig := tuples[0].Orig; orig != "Poyekhali!" {
+		t.Fatalf("Wrong Orig (%s), should be 'Hello, there!'", orig)
+	}
+}
```

Comment:
Why don't compare a datetime value here too?

Reply:
Thanks for reminder and following to changes in my pull request!

Reply:
No problem, but I still don't see a datetime value assert here...

Reply:
It will not be here until comment is unresolved.